lib/std/testing.zig

```diff
@@ -21,6 +23,77 @@ pub var base_allocator_instance = std.heap.FixedBufferAllocator.init("");
 /// TODO https://github.com/ziglang/zig/issues/5738
 pub var log_level = std.log.Level.warn;
 
+/// Non-zero counts imply test failure. The value is reset for every test.
+pub var fail_count: usize = 0;
```

Comment:
Tests are planned to be run in parallel, so this will be a race condition. I do neither understand, why counting is useful or necessary in unit tests. The whole point of a unit test is to abstract away the details unless the unit fails.

Reply:
Flagging a test as failed with an increment, and testing with a zero compare is as cheap as it gets. No strong feelings against `bool`, although I think the number in the final line may help some edge cases such as fixing one out of multiple causes at a time.

```
Test [1/10] test.parse and render IPv6 addresses... failed
> parse IP6 FF01:0:0:0:0:0:0:FB got print [ff01::fb]:0, want [ff01::fb]:0-makeitfail
> resolve IP6 FF01:0:0:0:0:0:0:FB got print [ff01::fb]:0, want [ff01::fb]:0-makeitfail
> parse IP6 FF01::Fb got print [ff01::fb]:0, want [ff01::fb]:0-makeitfail
> resolve IP6 FF01::Fb got print [ff01::fb]:0, want [ff01::fb]:0-makeitfail
> parse IP6 ::1 got print [::1]:0, want [::1]:0-makeitfail
> resolve IP6 ::1 got print [::1]:0, want [::1]:0-makeitfail
> parse IP6 :: got print [::]:0, want [::]:0-makeitfail
> resolve IP6 :: got print [::]:0, want [::]:0-makeitfail
> parse IP6 1:: got print [1::]:0, want [1::]:0-makeitfail
> resolve IP6 1:: got print [1::]:0, want [1::]:0-makeitfail
> parse IP6 2001:db8:: got print [2001:db8::]:0, want [2001:db8::]:0-makeitfail
> resolve IP6 2001:db8:: got print [2001:db8::]:0, want [2001:db8::]:0-makeitfail
> parse IP6 ::1234:5678 got print [::1234:5678]:0, want [::1234:5678]:0-makeitfail
> resolve IP6 ::1234:5678 got print [::1234:5678]:0, want [::1234:5678]:0-makeitfail
> parse IP6 2001:db8::1234:5678 got print [2001:db8::1234:5678]:0, want [2001:db8::1234:5678]:0-makeitfail
> resolve IP6 2001:db8::1234:5678 got print [2001:db8::1234:5678]:0, want [2001:db8::1234:5678]:0-makeitfail
> parse IP6 FF01::FB%1234 got print [ff01::fb]:0, want [ff01::fb]:0-makeitfail
> resolve IP6 FF01::FB%1234 got print [ff01::fb]:0, want [ff01::fb]:0-makeitfail
> parse IP6 ::ffff:123.5.123.5 got print [::ffff:123.5.123.5]:0, want [::ffff:123.5.123.5]:0-makeitfail
> resolve IP6 ::ffff:123.5.123.5 got print [::ffff:123.5.123.5]:0, want [::ffff:123.5.123.5]:0-makeitfail
Test [1/10] test.parse and render IPv6 addresses... FAIL (20 failures)
```